pkg/virt-api/webhooks/validating-webhook/admitters/pod-eviction-admitter.go

```diff
@@ -0,0 +1,84 @@
+package admitters
+
+import (
+	"fmt"
+	"net/http"
+
+	"k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	virtv1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+)
+
+type PodEvictionAdmitter struct {
+	ClusterConfig *virtconfig.ClusterConfig
+	VirtClient    kubecli.KubevirtClient
+}
+
+func (admitter *PodEvictionAdmitter) Admit(ar *v1beta1.AdmissionReview) *v1beta1.AdmissionResponse {
+	if !admitter.ClusterConfig.LiveMigrationEnabled() {
+		return allowed()
+	}
+
+	launcher, err := admitter.VirtClient.CoreV1().Pods(ar.Request.Namespace).Get(ar.Request.Name, metav1.GetOptions{})
+	if err != nil {
+		return allowed()
+	}
+
+	if value, exists := launcher.GetLabels()[virtv1.AppLabel]; !exists || value != "virt-launcher" {
+		return allowed()
+	}
+
+	domainName, exists := launcher.GetAnnotations()[virtv1.DomainAnnotation]
+	if !exists {
+		return allowed()
+	}
+
+	vmi, err := admitter.VirtClient.VirtualMachineInstance(ar.Request.Namespace).Get(domainName, &metav1.GetOptions{})
```

Comment:
We really need to be using an informer here rather than making GET calls within the api webhook. This goes for both the VMI and POD GET requests.

in your 

```
type PodEvictionAdmitter struct {
	ClusterConfig *virtconfig.ClusterConfig
	VirtClient    kubecli.KubevirtClient
}
```

struct, that ClusterConfig object contains informers. You simply need to add a vmi and vmi pod informer there which will be immediately accessible to this webhook code. That's the pattern we've already established for other webhooks.